src/Analyzers/CSharp/Analyzers/UseObjectInitializer/CSharpUseObjectInitializerDiagnosticAnalyzer.cs

```diff
@@ -31,5 +31,9 @@ protected override bool AreObjectInitializersSupported(SyntaxNodeAnalysisContext
         }
 
         protected override ISyntaxFacts GetSyntaxFacts() => CSharpSyntaxFacts.Instance;
+
+        protected override bool IsValidContainingStatement(StatementSyntax node)
+            => node is not LocalDeclarationStatementSyntax localDecl
+            || localDecl.UsingKeyword == default;
```

Comment:
Either:

```suggestion
            => node is not LocalDeclarationStatementSyntax localDecl ||
               localDecl.UsingKeyword == default;
```

Or:

```suggestion
            => node is not LocalDeclarationStatementSyntax localDecl
               || localDecl.UsingKeyword == default;
```

Plz.

Reply:
also, this code needs a good comment not only explaining *what* it is matching against, but why this pattern is important to be defensive against.

Without looking at the original bug, it wasn't clear to me why we wanted to block this for `using var ...` cases.  Now i understand, but that still isn't clear in the code itself (i..e that the initialization code may need ot run in the scope of the using block, and semantic coudl break if we lift out).

Reply:
I went the first option because I like the code lining. Surprised you're okay with a 3 space indent 😛